preflight_parser/preflight_parser/__init__.py

```diff
@@ -1033,7 +1089,7 @@ def compute_toplevel_files(roots: dict[str, ParsedTeXFile], nodes: dict[str, Par
             filename=n.filename,
             process=MainProcessSpec(compiler=CompilerSpec(engine=engine, output=output, lang=lang, postp=postprocess)),
         )
-        compiler: str | None = tl.process.compiler.compiler_string
+        compiler: str | None = None if tl.process.compiler is None else tl.process.compiler.compiler_string
```

Comment:
Not sure what is more readable for me ... the "if X is something falsish, return X, otherwise return Y" is common in Perl, but in Python I always stumble about "What was it again ... didn't `and` return True/False ..."

BTW, I think it was a failure to design Python like this - because it is Perl dialact ;-)